analyzers/src/SonarAnalyzer.CFG/ShimLayer/CompilationOptionsWrapper.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Lightup
+{
+    public readonly struct CompilationOptionsWrapper
+    {
+        internal const string WrappedTypeName = "Microsoft.CodeAnalysis.CompilationOptions";
+        private static readonly Type WrappedType;
+
+        private static readonly Func<CompilationOptions, object> SyntaxTreeOptionsProviderAccessor;
+        private readonly CompilationOptions node;
+
+        static CompilationOptionsWrapper()
+        {
+            WrappedType = WrapperHelper.GetWrappedType(typeof(CompilationOptionsWrapper));
+            SyntaxTreeOptionsProviderAccessor = LightupHelpers.CreateSyntaxPropertyAccessor<CompilationOptions, object>(WrappedType, nameof(SyntaxTreeOptionsProvider));
+        }
```

Comment:
I find initializing the fields directly more straightforward than a static constructor.

Reply:
So do I. I don't like most of the code. I'm just following the style that is kept in the ShimLayer (not our code base, but copy/paste from StyleCop repository)